src/containers/Category.js

```diff
@@ -33,59 +29,53 @@ function getListID(paramCategory) {
-  static fetchData({ params, store }) {
-    return store.dispatch(fetchListedPosts(getListID(params.category), categories, MAXRESULT))
-  }
-
-  constructor(props) {
-    super(props)
-    const category = this.props.params.category
-    this.state = {
-      catId: getListID(category)
+  // params are passed from Route component of react-router
+  static fetchData({ params, store, query }) {
+    /* fetch page 1 if page is invalid */
+    let page = parseInt(_.get(query, 'page', 1), 10)
+    if (isNaN(page) || page < 0) {
+      page = 1
     }
-    this.loadMore = this._loadMore.bind(this)
+    return store.dispatch(fetchListedPosts(getListID(params.category), categories, MAXRESULT, page))
   }
 
   componentWillMount() {
-    const { lists, fetchListedPosts } = this.props
-
-    let catId = this.state.catId
-
-    // if fetched before, do nothing
-    if (_.get(lists, [ catId, 'items', 'length' ], 0) > 0) {
-      return
-    }
-
-    fetchListedPosts(catId, categories, MAXRESULT)
+    const { fetchListedPosts, catId } = this.props
+    const page = _.get(this.props, 'page', 1)
 
+    fetchListedPosts(catId, categories, MAXRESULT, page)
   }
 
   componentWillReceiveProps(nextProps) {
-    const { lists, fetchListedPosts, params } = nextProps
-    let catId = getListID(params.category)
+    const { fetchListedPosts, catId } = nextProps
+    const page = _.get(nextProps, 'page', 1)
 
-    // if fetched before, do nothing
-    if (_.get(lists, [ catId, 'items', 'length' ], 0) > 0) {
-      return
-    }
-
-    fetchListedPosts(catId, categories, MAXRESULT)
-  }
-
-  _loadMore() {
-    const { fetchListedPosts, params } = this.props
-    let catId = getListID(params.category)
-    fetchListedPosts(catId, categories, MAXRESULT)
+    fetchListedPosts(catId, categories, MAXRESULT, page)
   }
 
   render() {
-    const { lists, entities, params } = this.props
-    const category = _.get(params, 'category', '')
+    const { lists, entities, catId, category, page, pathname } = this.props
     const postEntities = _.get(entities, reduxStateFields.postsInEntities, {})
-    const catId = getListID(category)
     const error = _.get(lists, [ catId, 'error' ], null)
+
+    // total items will be in that catId
     const total = _.get(lists, [ catId, 'total' ], 0)
-    const posts = utils.denormalizePosts(_.get(lists, [ catId, 'items' ], []), postEntities)
+
+    // pages will be like
+    // {
+    //   1: [0, 9],
+    //   3: [10, 19],
+    // }
+    //
+    // which means the items of page 1 are in items[0] to items[9],
+    // the items of page 3 are in items[10] to item [19]
+    const pages = _.get(lists, [ catId, 'pages' ], {})
+    const startPos = _.get(pages, [ page, 0 ], 0)
+    const endPos = _.get(pages, [ page, 1 ], 0)
+    // denormalize the items of current page
+    const posts = utils.denormalizePosts(_.get(lists, [ catId, 'items' ], []).slice(startPos, endPos + 1), postEntities)
+
+    const totalPages = total / MAXRESULT
```

Comment:
`const totalPages = Math.ceil(total / MAXRESULT)` will be safer since `Pagination` might only be able to handle integer.